internal/constants/scripts.go

```diff
@@ -133,3 +133,80 @@ function correlate_int_mac() {
 
 correlate_int_mac
 `
+
+// PreNetworkConfigScript script runs on hosts before network manager service starts in order to copy
+// the host specifics *.nmconnection files to /NetworkManager/system-connections/ based on host MAC-address matching
+const PreNetworkConfigScript = `
+#!/bin/bash
+
+#TODO: set file path for unified nmconnection file of static conf
+unified_nmconnections=/tmp/nmconn
+if [ -z "$unified_nmconnections" ]
+then
+  exit 0
+fi
+
+# array storing all host mac addresses
+host_macs=()
+
+# temp dir storing connection files
+conn_temp_dir=$(mktemp -d)
+
+# the path of the connection file matching any of host's mac addresses
+matched_connection_file=""
+
+# list host mac addresses for active network interfaces
+function list_macs() {
+  SYS_CLASS_NET_DIR='/sys/class/net'
+  for nic in $( ls $SYS_CLASS_NET_DIR )
+  do
+    if grep -q up $SYS_CLASS_NET_DIR/$nic/operstate
+    then
+      mac=$(cat $SYS_CLASS_NET_DIR/$nic/address | tr '[:lower:]' '[:upper:]')
+      host_macs+=($mac)
+    fi
+  done
+}
+
```

Comment:
General comment: you don't need to split the unified file. Ignition will already contain an nmconnection file per MAC.

Reply:
fixed